Makefile

```diff
@@ -15,51 +15,59 @@ else
 	COMPOSER_ARGS := --classmap-authoritative
 	ZIP_NAME := bmlt-root-server-3.0.0-build$(GITHUB_RUN_NUMBER)-$(GITHUB_SHA).zip
 endif
-
 BASE_IMAGE := public.ecr.aws/bmlt/bmlt-root-server-base:latest
+CROUTON_JS := src/legacy/client_interface/html/croutonjs/crouton.js
+VENDOR_AUTOLOAD := src/vendor/autoload.php
+ZIP_FILE := build/bmlt-root-server.zip
+
+$(VENDOR_AUTOLOAD):
+	docker run $(DOCKER_ARGS) -v $(shell pwd)/src:/app -w /app $(BASE_IMAGE) composer install $(COMPOSER_ARGS)
+
+$(CROUTON_JS):
+	curl -sLO https://github.com/bmlt-enabled/crouton/releases/latest/download/croutonjs.zip
+	mkdir -p src/legacy/client_interface/html/croutonjs
+	unzip croutonjs.zip -d src/legacy/client_interface/html/croutonjs
+	rm -f croutonjs.zip
+	rm -f src/legacy/client_interface/html/croutonjs/*.html
+	rm -f src/legacy/client_interface/html/croutonjs/*.json
+	rm -rf src/legacy/client_interface/html/croutonjs/examples
+
+$(ZIP_FILE): $(VENDOR_AUTOLOAD) $(CROUTON_JS)
+	mkdir -p build
+	cp -r src build/main_server
+	cd build && zip -r $(shell basename $(ZIP_FILE)) main_server
+	rm -rf build/main_server
 
-.PHONY: help
 help:  ## Print the help documentation
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-build: clean composer-install crouton-install build-docker ## Builds Full Image
+.PHONY: composer
+composer: $(VENDOR_AUTOLOAD) ## Runs composer install
 
-build-zip: clean composer-install crouton-install zip-build ## Builds Full Zip
+.PHONY: crouton
+crouton: $(CROUTON_JS) ## Installs crouton
 
-.PHONY: build-docker
-build-docker:  ## Builds Docker Image
+.PHONY: docker
+docker: $(VENDOR_AUTOLOAD) $(CROUTON_JS) ## Builds Docker Image
 	docker build -f docker/$(DOCKERFILE) . -t $(IMAGE):$(TAG)
 
-.PHONY: zip-build
-zip-build:  ## Zips the build
-	rm -rf build
-	mkdir -p build
-	cp -r src build/main_server
-	cd build && zip -r $(ZIP_NAME) main_server && cd ..
+.PHONY: zip
+zip: $(ZIP_FILE) ## Builds zip file
 
-.PHONY: composer-install
-composer-install:  ## Composer Install
-	docker run $(DOCKER_ARGS) -v $(shell pwd)/src:/app -w /app $(BASE_IMAGE) composer install $(COMPOSER_ARGS)
+.PHONY: docker-push
+docker-push: ## Pushes docker image to ECR
+	docker push $(IMAGE):$(TAG)
 
-.PHONY: crouton-install
-crouton-install:   ## Installs Crouton
-	curl -sLO https://github.com/bmlt-enabled/crouton/releases/latest/download/croutonjs.zip
-	mkdir -p src/legacy/client_interface/html/croutonjs
-	unzip croutonjs.zip -d src/legacy/client_interface/html/croutonjs
-	rm croutonjs.zip
-	rm src/legacy/client_interface/html/croutonjs/*.html
-	rm src/legacy/client_interface/html/croutonjs/*.json
-	rm -rf src/legacy/client_interface/html/croutonjs/examples
+.PHONY: ecr-login
+ecr-login: ## Authenticates to ECR
+	aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/t5y4k5q3
```

Comment:
we can make this public.ecr.aws/bmlt
this was the registry name before the accepted our alias and we moved everything in small span